client/Windows/wf_graphics.c

```diff
@@ -237,24 +237,35 @@ static BOOL wf_Pointer_New(rdpContext* context, const rdpPointer* pointer)
 		            pointer->height);
 		info.hbmMask = CreateBitmap(pointer->width, pointer->height, 1, 1, pdata);
 		_aligned_free(pdata);
-		pdata = (BYTE*) _aligned_malloc(pointer->width * pointer->height *
-		                                GetBitsPerPixel(gdi->dstFormat), 16);
 
-		if (!pdata)
+		UINT32 srcFormat;
+		switch (pointer->xorBpp)
+		{
+		case 16:
```

Comment:
Is this always 15bpp?
Case for 8bpp and 15bpp is missing or are they not supported at all?

Reply:
Well 8bpp was falling through to palette format below. This switch is just a reverse map of what's in the freerdp_image_copy_from_pointer_data() routine, and other resolutions aren't supported by it. I should be more specific in this case statement. 

But, can I just get rid of this case statement and using  freerdp_image_copy_from_pointer_data altogether? Do you know why it's even in there. The and and xor data should already be in a usable format for CreateBitmap and CreateDIBSection (other than flipping it to a top-down bitmap). I should be able to call flip_bitmap on the xor data and pass it directly to CreateDIBSection instead

Reply:
I will look into what bit depths are supported for color cursors, maybe some <24bpp formats do need to be converted to an RGB (24 or 32) format. I'll also check what happens in the 16bpp case

Reply:
@perkerk You're probably right, the `freerdp_image_copy_from_pointer_data` is used on non windows platforms to convert to usable bitmap formats.